online-shop/src/app/components/containers/product-list/product-list.component.ts

```diff
@@ -1,12 +1,28 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 import { products } from 'src/app/mocks/products.mocks';
 import { Product } from 'src/app/modules/shared/types/products.types';
+import { HttpClient } from '@angular/common/http';
+import { Observable, of } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { ProductService } from 'src/app/services/products.service';
 @Component({
   selector: 'app-product-list',
   templateUrl: './product-list.component.html',
   styleUrls: ['./product-list.component.scss'],
 })
-export class ProductListComponent {
-  products: Product[] = products;
-  
+export class ProductListComponent implements OnInit{
+  products: Product[] = [];
+
+  constructor(private productService: ProductService) {
+    this.getProducts();
+  }
+
+  ngOnInit(): void {
+    this.getProducts();
+  }
+  getProducts(): void {
+    this.productService
+      .getAllProducts()
+      .subscribe((prod) => (this.products = prod));
```

Comment:
Make sure that you unsubscribe from each subscription you do in the app. Check the following [https://github.com/ngneat/until-destroy](https://github.com/ngneat/until-destroy)

Reply:
Used @UntilDestroy({ checkProperties: true }).